OpenRA.Mods.Common/Widgets/WorldCommandWidget.cs

```diff
@@ -191,15 +191,24 @@ bool CycleBases()
 				.Select(b => b.Actor)
 				.ToList();
 
-			if (!bases.Any())
-				return true;
+            // If no base is found all existing buildings are selected.
+            if (!bases.Any())
+            {
+				bases = world.ActorsWithTrait<TargetableBuilding>()
+                .Where(a => a.Actor.Owner == world.LocalPlayer)
+                .Select(b => b.Actor)
+                .ToList();
+                if (!bases.Any())
+					return true;
+            }   
 
 			var next = bases
 				.SkipWhile(b => !world.Selection.Actors.Contains(b))
 				.Skip(1)
 				.FirstOrDefault();
 
-			if (next == null)
+            // If no construction yard exists then next is set to only the oldest existing building
+			if (next == null || !next.HasTrait<BaseBuilding>())
```

Comment:
Why is this? From what I see, this change isn't needed. bases will contain all buildings if no conyards and only conyards when conyards, since you reused the variable above. This check was fine as-is.